liblumen_codegen/lib/lumen/compiler/Dialect/EIR/Conversion/EIRToLLVM/ConvertEIRToLLVM.cpp

```diff
@@ -554,7 +559,7 @@ struct CondBranchOpConversion : public EIROpConversion<eir::CondBranchOp> {
       if (maskInfo.requiresShift()) {
         Value shiftConst =
             llvm_constant(termTy, getIntegerAttr(rewriter, maskInfo.shift));
-        Value shiftedCond = llvm_shl(maskedCond, shiftConst);
+        Value shiftedCond = llvm_shr(maskedCond, shiftConst);
```

Comment:
ooff... that must have been an annoying bug to find.

Reply:
I got lucky and caught this before it actually became a bug, but yeah, would've been super annoying